### ISSUE TITLE
ToggleSwitch state isn't displayed properly after Styles change

### ISSUE THREAD

**Issue by Splitwirez** [AvaloniaUI/Avalonia#5379]
**Describe the bug**
The ToggleSwitch control's grip sometimes appears in the "IsChecked=False" position even though the control's `IsChecked` property isn't `False` after the applicable Styles change (likely the `Template`).

**To Reproduce**
Build and run the repro I've prepared here: [ToggleSwitchGripPositionRepro.zip](https://github.com/AvaloniaUI/Avalonia/files/5890134/ToggleSwitchGripPositionRepro.zip)

**Expected behavior**
The ToggleSwitch's grip remains on in the "IsChecked=True" position.

**Desktop (please complete the following information):**
 - OS: Kubuntu Linux
 - Version 20.04

**Additional context**
Part of me wonders if this may actually be my fault...I submitted a PR regarding ToggleSwitches which made the grip draggable (as is literally the defining feature of a ToggleSwitch), and part of me wonders if that may have brought this about...

**Comment by jinyuttt** [comment]
Avalonia.Ribbon ，Did you give up?

**Comment by Splitwirez** [comment]
> Avalonia.Ribbon ，Did you give up?
No, far from it. I was just working on AvaloniaRibbon earlier this week, and intend to continue doing so.

That has nothing to do with this ToggleSwitch problem though...

**Comment by newuser** [comment]
Issue reproduces. Bounds were 0 for toggleswitch after click and knob position is related to it.
